modules/persistent-storage-csi-snapshots-create.adoc

```diff
@@ -0,0 +1,127 @@
+
+.Procedure
+
+To dynamically create a volume snapshot:
+
+. Save your volume snapshot class definition to a file.
+
++
+.volumesnapshotclass.yaml
+[source,yaml]
+----
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshotClass <1>
+metadata:
+  name: csi-hostpath-snap
+driver: hostpath.csi.k8s.io
+deletionPolicy: Delete
+----
+<1> Specifies the attributes that point to a CSI driver, such as HostPath.
++
+. Create the object definition file you saved in the previous step.
++
+----
+$ oc create -f volumesnapshotclass.yaml
+----
+
+. Create a VolumeSnapshot object:
+
++
+.volumesnapshot-dynamic.yaml
+[source,yaml]
+----
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot <1>
+metadata:
+  name: mysnap
+spec:
+  volumeSnapshotClassName: csi-hostpath-snap <2>
+  source:
+    persistentVolumeClaimName: myclaim <3>
+----
+<1> A VolumeSnapshot object is automatically created based on the VolumeSnapshot.
++
+<2> The request for a particular class by the volume snapshot. If nothing is set, the default class is used if available.
++
+<3> The name of the PersistentVolumeClaim bound to a PersistentVolume. This defines what you want to create a snapshot of. Required for dynamically provisioning a snapshot.
+
+. Create the object definition file you saved in the previous step:
++
+----
+$ oc create -f volumesnapshot<provisioning-method>.yaml
+----
+
+
+To manually provision a snapshot:
+
+. Specify a volumeSnapshotContentName as the source for the snapshot, in addition to defining volume snapshot class and PVC name as shown above.
++
+.volumesnapshot-manual.yaml
+[source,yaml]
+----
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot
+metadata:
+  name: snapshot-demo
+spec:
+  source:
+    volumeSnapshotContentName: mycontent <1>
+----
+<1> The volumeSnapshotContentName source field is required for pre-provisioned snapshots.
+
+. Create the object definition file you saved in the previous step:
++
+----
+$ oc create -f volumesnapshot<provisioning-method>.yaml
+----
+
+Once the snapshot has been created in the cluster, additional details about the snapshot are available.
+
+. Display details about the volume snapshot that was created.
++
+----
+$ oc describe volumesnapshot<provisioning-method> mysnap
+----
+
+The following example displays additional details about the volume snapshot you created.
+
+.volumesnapshot.yaml
+[source,yaml]
+----
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshot
+metadata:
+  name: mysnap
+spec:
+  source:
+    persistentVolumeClaimName: myclaim
+  volumeSnapshotClassName: csi-hostpath-snap
+status:
+  boundVolumeSnapshotContentName: snapcontent-1af4989e-a365-4286-96f8-d5dcd65d78d6 <1>
+  creationTime: "2020-01-29T12:24:30Z"
+  readyToUse: true
+  restoreSize: 500Mi
+----
+<1> The volume snapshot content is the pointer to the storage itself.
+
+To verify that the volume snapshot was created automatically, run the following command:
```

Comment:
Once `boundVolumeSnapshotContentName` is populated it means that there's a corresponding VolumeSnapshotContent. Users can confirm that the VolumeSnapshot has `readyToUse: true` to ensure that the snapshot is ready.